File: Easy/BubbleSort.py

def bubbleSort(array):
    isSorted = False
    count = 0
    while not isSorted:
        isSorted = True
        for i in range(len(array) - 1 - count):
            if array[i+1] > array[i]:
                temp = array[i+1]
                array[i+1] = array[i]
                array[i] = temp
                isSorted = False
        count += 1
    return array

File: Easy/BubbleSort.cpp
#include<iostream>
#include<vector>
using namespace std;

void bubbleSort(vector<int> &arr, int len)
{
    bool notSorted = true;
    while(true)
    {
        notSorted = false;
        for(int i=0; i<len-1; i++)
        {
            if(arr[i]>arr[i+1])
            {
                swap(arr[i],arr[i+1]);
                notSorted = true;
            }
        }
        if(!notSorted)
        {
            break;
        }
    }
}

int main()
{
    int len,temp;
    vector<int> arr;
    cin>>len;
    for(int i=0; i<len; i++)
    {
        cin>>temp;
        arr.push_back(temp);
    }
    bubbleSort(arr, len);
    cout<<"sorted:";
    for(int i=0; i<len; i++)
    {
        cout<<arr[i]<<" ";
    }
    return 0;
}